### PR TITLE
Handle when VSO employee can't access file number

### PR DESCRIPTION
**Pull request by monfresh**
**Why**: The controller design was resulting in a double render error
`AbstractController::DoubleRenderError` in the scenario where a VSO
employee doesn't have access to the file number. This scenario was not
tested.

This PR adds tests for that scenario, and others that weren't tested,
and also cleans up the controller by moving the business logic in
separate classes.

Fixes Sentry errors such as:
https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/3663/events/241077/

Resolves #{github issue number}

### Acceptance Criteria
- When signed in as a VSO employee, searching for a claims file number that you don't have access to doesn't result in a `AbstractController::DoubleRenderError` server error.

### Testing Plan
1. Sign in as a VSO employee
2. Search for a claims file number that you don't have access to
Expected Result: you see a helpful error message that says "You don't have access to this claims file number"

### CODE REVIEW

File: app/workflows/appeals_for_file_number.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  private
+
+  attr_reader :file_number, :user, :appeals
+
+  def appeals_for_file_number
+    MetricsService.record("VACOLS: Get appeal information for file_number #{file_number}",
+                          service: :queue,
+                          name: "AppealsController.index") do
```
[the remark this answers is not in the document]
Comment (monfresh) [reply app/workflows/appeals_for_file_number.rb]: yeah, I meant to ask about that. I will change it.

**Comment by monfresh** [comment]
Before, generic 500 errors resulted in this error message to the user:
screen shot 2019-02-28 at 3 06 23 pm

With the changes in this PR, the error now is more helpful:
screen shot 2019-03-01 at 12 05 23 pm

**Comment by lowellrex** [comment]
I think I'm a little out of my depth here with this new workflows/`FormResponse` pattern. Do we have a tech spec that explains this pattern so I can understand it better before commenting on this PR?

**Comment by monfresh** [comment]
@lowellrex Tech Spec here: https://github.com/department-of-veterans-affairs/caseflow/issues/9775. I'd love to hear your thoughts, and let me know if you have any questions.

**Pull request merged**